File: CustomOnboardingIntro/app/src/main/java/nisrulz/github/example/customonboardingintro/IntroFragment.java
package nisrulz.github.example.customonboardingintro;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class IntroFragment extends Fragment {

  private static final String BACKGROUND_COLOR = "backgroundColor";
  private static final String PAGE = "page";

  private int mBackgroundColor, mPage;

  public static IntroFragment newInstance(int backgroundColor, int page) {
    IntroFragment frag = new IntroFragment();
    Bundle b = new Bundle();
    b.putInt(BACKGROUND_COLOR, backgroundColor);
    b.putInt(PAGE, page);
    frag.setArguments(b);
    return frag;
  }

  @Override public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    if (!getArguments().containsKey(BACKGROUND_COLOR)) {
      throw new RuntimeException("Fragment must contain a \"" + BACKGROUND_COLOR + "\" argument!");
    }
    mBackgroundColor = getArguments().getInt(BACKGROUND_COLOR);

    if (!getArguments().containsKey(PAGE)) {
      throw new RuntimeException("Fragment must contain a \"" + PAGE + "\" argument!");
    }
    mPage = getArguments().getInt(PAGE);
  }

  @Override public View onCreateView(LayoutInflater inflater, ViewGroup container,
      Bundle savedInstanceState) {

    // Select a layout based on the current page
    int layoutResId;
    switch (mPage) {
      case 0:
        layoutResId = R.layout.fragment_intro1;
        break;
      default:
        layoutResId = R.layout.fragment_intro2;
    }

    // Inflate the layout resource file
    View view = getActivity().getLayoutInflater().inflate(layoutResId, container, false);

    // Set the current page index as the View's tag (useful in the PageTransformer)
    view.setTag(mPage);

    return view;
  }

  @Override public void onViewCreated(View view, Bundle savedInstanceState) {
    super.onViewCreated(view, savedInstanceState);

    // Set the background color of the root view to the color specified in newInstance()
    View background = view.findViewById(R.id.intro_background);
    background.setBackgroundColor(mBackgroundColor);
  }
}

File: CustomOnboardingIntro/app/src/main/java/github/nisrulz/example/customonboardingintro/IntroFragment.kt
package github.nisrulz.example.customonboardingintro

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment

class IntroFragment : Fragment() {

    private var mBackgroundColor = 0
    private var mPage = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.apply {
            if (!containsKey(BACKGROUND_COLOR)) {
                throw RuntimeException("Fragment must contain a \"$BACKGROUND_COLOR\" argument!")
            }
            mBackgroundColor = getInt(BACKGROUND_COLOR)
            if (!containsKey(PAGE)) {
                throw RuntimeException("Fragment must contain a \"$PAGE\" argument!")
            }
            mPage = getInt(PAGE)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        // Select a layout based on the current page
        val layoutResId: Int = when (mPage) {
            0 -> R.layout.fragment_intro1
            else -> R.layout.fragment_intro2
        }

        // Inflate the layout resource file
        val view = activity?.layoutInflater?.inflate(layoutResId, container, false)

        // Set the current page index as the View's tag (useful in the PageTransformer)
        view?.tag = mPage
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // Set the background color of the root view to the color specified in newInstance()
        val background = view.findViewById<View>(R.id.intro_background)
        background.setBackgroundColor(mBackgroundColor)
    }

    companion object {
        private const val BACKGROUND_COLOR = "backgroundColor"
        private const val PAGE = "page"

        fun newInstance(backgroundColor: Int, page: Int): IntroFragment {
            val frag = IntroFragment()
            val b = Bundle()
            b.putInt(BACKGROUND_COLOR, backgroundColor)
            b.putInt(PAGE, page)
            frag.arguments = b
            return frag
        }
    }
}